Added Drozone to favorites quickaccess

newhinton opened the pull request:
closes #10908
This adds the ability to drop a folder on the quickaccess and add it to it

newhinton committed:
Extract variable for yellow color in icons

Signed-off-by: fnuesse <felix.nuesse@t-online.de>

newhinton committed:
Fix drop zone shadow

Signed-off-by: fnuesse <felix.nuesse@t-online.de>

newhinton committed:
Add drop zone for favorites quick access in navigation

Signed-off-by: fnuesse <felix.nuesse@t-online.de>

newhinton commented:
CI is failing, but i think the errors are unrelated:
```  
Scenario: create a new user with a custom display name                       # /drone/src/github.com/nextcloud/server/tests/acceptance/features/users.feature:12
    Given I am logged in as the admin                                          # LoginPageContext::iAmLoggedInAsTheAdmin()
    And I open the User settings                                               # SettingsMenuContext::iOpenTheUserSettings()
    When I click the New user button                                           # UsersSettingsContext::iClickTheNewUserButton()
    And I see that the new user form is shown                                  # UsersSettintestgsContext::iSeeThatTheNewUserFormIsShown()
    And I set the user name for the new user to "test"                         # UsersSettingsContext::iSetTheUserNameForTheNewUserTo()
    And I set the display name for the new user to "Test display name"         # UsersSettingsContext::iSetTheDisplayNameForTheNewUserTo()
    And I set the password for the new user to "123456acb"                     # UsersSettingsContext::iSetThePasswordForTheNewUserTo()
    And I create the new user                                                  # UsersSettingsContext::iCreateTheNewUser()
    Then I see that the list of users contains the user "test"                 # UsersSettingsContext::iSeeThatTheListOfUsersContainsTheUser()
    And I see that the display name for the user "test" is "Test display name" # UsersSettingsContext::iSeeThatTheDisplayNameForTheUserIs()
      Row for user test in Users Settings could not be found after 100 seconds
      displayName cell for user test in Users Settings could not be found after 100 seconds
      displayName input for user test in Users Settings could not be found after 100 seconds (NoSuchElementException)
sh: 1: kill: No such process

```

and
```

Enable an app bundle                                          # /drone/src/github.com/nextcloud/server/tests/acceptance/features/apps.feature:42
    Given I act as Jane                                                   # ActorContext::iActAs()
    And I am logged in as the admin                                       # LoginPageContext::iAmLoggedInAsTheAdmin()
    And I open the Apps management                                        # SettingsMenuContext::iOpenTheAppsManagement()
    And I open the "App bundles" section                                  # AppNavigationContext::iOpenTheSection()
    When I enable all apps from the "Enterprise bundle"                   # AppsManagementContext::iEnableAllAppsFromThe()
    Then I see that the "Auditing / Logging" app has been enabled         # AppsManagementContext::iSeeThatTheAppHasBeenEnabled()
      Disable button in the app list for Auditing / Logging could not be found after 100 seconds (NoSuchElementException)
And I see that the "LDAP user and group backend" app has been enabled # AppsManagementContext::iSeeThatTheAppHasBeenEnabled()

```

danxuliu commented on apps/files/js/filelist.js:
```diff
@@ -349,10 +349,16 @@
 			$.event.trigger({type: "droppedOnTrash"});
 
-			var self=this;
+			var _self=this;
```
Why the rename?

newhinton replied:
Initially i had problems with self, it would not work properly. I can look if it works now

danxuliu commented on apps/files/js/filelist.js:
```diff
@@ -349,10 +349,16 @@
+				_self.do_delete(filename, directory);
+			});
+
+			$.event.trigger({type: "droppedOnFavorites"});
```
Why is this needed?

newhinton replied:
This deletes the folder when it is dropped on the trash

danxuliu commented on apps/files/js/filelist.js:
```diff
@@ -349,10 +349,16 @@
+			});
+
+			$.event.trigger({type: "droppedOnFavorites"});
+			_self=this;
```
`_self` was not modified since it was set, so it is not needed to set it again.

newhinton replied:
You are right, this is a leftover because i implemented the trash dropzone and the favorites dropzone seperately and needed to make them seperate

danxuliu reviewed (commented):
Please rebase on current master :-) (then you will need to force push the branch again).

Tested and works, but it should be tested again once rebased ;-)

Maybe it would be good to not enable the favorite drop zone when dragging an already favorited folder, but that is a minor detail. In any case, a really nice feature, thanks :-D

danxuliu commented:
>  CI is failing, but i think the errors are unrelated:

Yes, `sh: 1: kill: No such process` is a crash in the PHP built-in server (do not worry, you did not cause it :-P ), and the other failure was caused by an app not having been released yet in the app store when the tests were run (again, nothing that you need to worry about ;-) ).

danxuliu committed:
Remove hack to prefetch the starred trash icon

Since 6ad7f32938 SVG icons are directly embedded in "icons-vars.css", so
the starred trash icon is now loaded along with the regular trash icon
all at once. Therefore it is not needed to explicitly prefetch it using
a hidden div.

Signed-off-by: Daniel Calviño Sánchez <danxuliu@gmail.com>

danxuliu committed:
Remove leftover from a WIP

Signed-off-by: Daniel Calviño Sánchez <danxuliu@gmail.com>

danxuliu committed:
Remove duplicated variable declaration

Signed-off-by: Daniel Calviño Sánchez <danxuliu@gmail.com>

danxuliu committed:
Remove unneeded triggering of event

The "droppedOnTrash" event was being triggered when the file list was
initialized, but it should be triggered only when the user actually
drops a file on the trash bin.

Besides that, the event had no effect; only the file list handles it,
but as it was not triggered on any element it ended being triggered on
the document, and thus not handled. Moreover, even if it had been
triggered on the file list it would have been done before the handler
was set, so it would not have been handled anyway. And even if it had
been handled no data was provided, so the handler would have failed.

In conclusion, triggering the event there was not needed, and thus it
was removed.

Signed-off-by: Daniel Calviño Sánchez <danxuliu@gmail.com>

danxuliu committed:
Add missing trailing ";"

Signed-off-by: Daniel Calviño Sánchez <danxuliu@gmail.com>

danxuliu committed:
Replace double quotes with single quotes

Signed-off-by: Daniel Calviño Sánchez <danxuliu@gmail.com>

danxuliu committed:
Make code format more consistent with the rest of the file

Signed-off-by: Daniel Calviño Sánchez <danxuliu@gmail.com>

danxuliu committed:
Use full names instead of abbreviations in variable names

Signed-off-by: Daniel Calviño Sánchez <danxuliu@gmail.com>

danxuliu committed:
Unify handling of dropping one file or several files on the trash bin

When a single file was dropped on the trash bin the file information was
gotten from the original element in the file list. When several files
were dropped on the trash bin the file information was gotten from the
helper elements being dragged around. The helper element also contain
the needed file information when a single file is being dragged, so the
handling was unified to always get the file information from the helper
elements.

As the handling of several files is the same as before there is still
the issue of only deleting those files shown in the drag helper instead
of all the selected files.

Signed-off-by: Daniel Calviño Sánchez <danxuliu@gmail.com>

danxuliu commented:
@newhinton, sorry for the delay.

I have taken the liberty of rebasing again your pull request and do some cleaning of the drop zone feature and the pull request history. Please refer to each individual commit and its comment for further information.

Regarding the commit that adds the drop zone to the favorites I have removed `.attr('data-favorite', elem.tags.includes("_$!<Favorite>!$_"))` (which should have been `.attr('data-favorite', elem.tags.includes(OC.TAG_FAVORITES))` ;-) ) because it was not needed with the current handling of the drop on the favorites and because, due to bugs in other parts of the Files code, `elem` may not always have a `tags` attribute (even if it should), which broke dragging rows in the file list (this could happen, for example, when using the _Select all_ checkbox).

danxuliu reviewed (approved):
Tested and works :+1:

Currently favoriting by dropping is limited to only one file or folder at a time due to some unrelated bugs in the Files app; I would merge this anyway in its current state.

Thanks again @newhinton!

MorrisJobke reviewed (approved):
Tested and works 👍

newhinton commented:
I didn't know that there was an OC.Tag introduced, i wrote a majority of this back at the conf, so there may have been some changes which im not aware of ^^ but good to know :D

danxuliu commented:
No problem at all about that @newhinton, you did not have to know it; I just mentioned it for your information :-)
